src/app/modules/shared_models/contacts_utils.nim

```diff
@@ -5,4 +5,8 @@ proc resolvePreferredDisplayName*(localNickName: string, ensName: string, displa
       return ensName
     if displayName != "":
       return displayName
-    return alias
+    if alias != "":
+      return alias
+    # This makes sure that people with no name are sorted last
+    # This fake name is never shown to the user
+    return "zzz"
```

Comment:
Ugh :) This should be handled at the QML side

Reply:
Hint: in my locale, "z" is not the last letter in the alphabet

Reply:
We should totally refrain from sorting on the NIM side; we already handle that in QML

Reply:
The sorting is indeed done on the QML side. However, `resolvePreferredDisplayName` is called for the `preferredDisplayName` property, which is the one used for the SFPM sorting. Since the `alias` is empty for all unknown contacts by default, it made it so that all unknown contacts sorted first.

So if I moved the `alias` resolution in the Item, then all members would evaluate the `alias` and we would lose the speed up.

If there is a way to force the ordering of unknown contacts to the bottom without this "hack", I'm very open.

Reply:
Hmm, ok, keep it for now :)